pkg/providers/vsphere/vsphere_test.go

```diff
@@ -2094,42 +2094,37 @@ func TestSetupAndValidateCreateClusterTemplateDoesNotExist(t *testing.T) {
 	tt := newProviderTest(t)
 
 	tt.setExpectationForSetup()
-	tt.setExpectationsForDefaultDiskGovcCalls()
-	tt.setExpectationForVCenterValidation()
-	tt.setExpectationsForMachineConfigsVCenterValidation()
-	tt.govc.EXPECT().SearchTemplate(tt.ctx, tt.datacenterConfig.Spec.Datacenter, tt.machineConfigs[tt.clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name]).Return("", nil)
+	tt.govc.EXPECT().SearchTemplate(tt.ctx, gomock.Any(), gomock.Any()).Return("", nil)
```

Comment:
Why do we need to use `gomock.Any()`?

Reply:
because it makes the code simpler, but I chawed it to look though the machines and add expectations for each machineconfig object